rfcs/0003-build-contents-diff.md

```diff
@@ -0,0 +1,152 @@
+- As a DevOps engineer or App Developer, when I am told there is a problem with a specific build (new build failing in CI or old build failing in production), I want to know what’s different about the problem build than the last one that worked. 
+- As an App Developer, I want to see the difference between the commit I built locally with pack yesterday and the build that kpack built for me today based on my commit. 
+ 
+Implications: 
+- Because the user will need to compare their most recent build or image to the build in production which is older, builds being compared will be non-adjacent 
+- Because the builds are non-adjacent, many things are likely to have changed in the contents and there will have been many “Reasons” that caused the intervening builds. Therefore, it doesn’t make a lot of sense to print the “Reason” as part of the diff
+- I assume that it is easy / possible for users to access the tag or digest associated with a container in production (perhaps there is another build identifier we could use?)
+
+
+## Prior Art 
+Differences from the reason diff RFC that’s been added to `kp build status`:
+- COMMIT can be depicted in the same way
+- CONFIG is the same
+- BUILDPACK is different because it prints the Bill of Materials which represents the results of the buildpacks used instead of only the buildpack/s that caused the new build to kick off
+- STACK is the same now, but will be different once we have more stack metadata such as the dpackage list (and someday a CVE list) that we can print
+ 
+Competitor examples: 
+https://github.com/GoogleContainerTools/container-diff
+
+## Complexity/Risks
+- Would this be helpful for diagnosing failed builds or builds that don't start? If so, how can we surface this info without a BoM being generated? 
+- How does all this show up in CI, if at all? 
+
+## Mockups 
+- Lines preceded by + or - will be bolded and colored green or red respectively 
+- Added a table as a header to the diff output that gives context on name, date/time, and tags for the two images (in case you diff builds from two different images)
+- CONFIG CHANGES includes the 5 lines above and below the diff
+- BILL OF MATERIALS (BUILDPACKS) CHANGES prints only info on the buildpacks that have changed 
+- STACK CHANGES includes a diff of mix-ins if they change version names
+ 
+------------------
+``` 
+$kp build diff petclinic index.docker.io/username/instance@sha256:fsdgh39478th0g9tuisodfhgvns938e54iunwuiehfasf9wgiauwsfx
+
+IMAGE NAME        BUILD TIME                 LATEST IMAGE 
+petclinic         2020-08-04 10:12:46        index.docker.io/username/instance@sha256:afdsniuubadsiugbn3784htw70835qioresgkdjfvba3493w8uiebgd
+petclinic         2020-09-08 12:08:36        index.docker.io/username/instance@sha256:fsdgh39478th0g9tuisodfhgvns938e54iunwuiehfasf9wgiauwsfx
+
+------------------------------
+
+COMMIT CHANGES
```

Comment:
The [source metadata ](https://github.com/buildpacks/spec/blob/313078611d8a7925cb69a241df58e7d749d7f364/platform.md#iobuildpacksprojectmetadata-json)available on CNB built images is available on `io.buildpacks.project.metadata` (Which may include source metadata).  It might make sense to model the output from that information.